static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -173,7 +173,12 @@ function WidgetBuilder({
     location.query.defaultWidgetQuery
   );
 
-  const isEditing = defined(widgetIndex);
+  const isEditing = widgetIndex !== NEW_ID;
+  const widgetIndexNum: number = Number(widgetIndex);
```

Comment:
nit: I think you can probably remove the explicit type since `Number()` already takes care of that for you
```suggestion
  const widgetIndexNum = Number(widgetIndex);
```